src/StreamDeck/Transport/LibUSBHIDAPI.py

```diff
@@ -57,7 +75,7 @@ def _load_hidapi_library(self, library_search_list):
             # The ctypes loader won't look in there by default unless the user has a Homebrew installed python, which gets patched
             # on installation.
             if not self.HOMEBREW_PREFIX:
-                type(self).HOMEBREW_PREFIX = self._get_homebrew_path()
+                self.HOMEBREW_PREFIX = self._get_homebrew_path()
```

Comment:
The `type(self).*` is intentional in this function, it allows us to search for the path on the first use of any class instance, then re-use that in any future instance until the program terminates (i.e. it's a static cache).

Here's a simple example of what it's doing:

```class Cache:
	CACHE = None

	def get(self):
		if not self.CACHE:
			print("Filling cache!")
			type(self).CACHE = "Blah"
		else:
			print("Re-using cache!")

		return self.CACHE


c = Cache()
c.get()

c2 = Cache()
c2.get()
```

If you run this as-is, you'll see despite it using two different Cache instances, it'll re-use the same value between them. If you change the `type(self)` to just `self`, the cached value is set on the local instance rather than the static class value and the cache will be initially empty on the second instance.

Reply:
In your example this is only working because you assign a immutable object to CACHE. If you use an immutable this is different.

```python
class Cache:
    CACHE = {}

    def get(self):
        if not self.CACHE:
            print("Filling cache!")
            self.CACHE['yolo'] = "Blah"
        else:
            print("Re-using cache!")

        return self.CACHE


c = Cache()
c.get()

c2 = Cache()
c2.get()
```

prints
```
Filling cache!
Re-using cache!
{'yolo': 'Blah'}
```

There are better and more standard ways to accomplish what you want to do though.